test/compress/dead-code.js

```diff
@@ -230,3 +230,25 @@ accessor: {
     }
     expect: {}
 }
+
+issue_2233: {
+    options = {
+        pure_getters: "strict",
+        side_effects: true,
+        unsafe: true,
+    }
+    input: {
+        Array.isArray;
+        Boolean;
+        console.log;
+        Error.name;
+        Function.length;
+        Math.random;
+        Number.isNaN;
+        RegExp;
+        Object.defineProperty;
+        String.fromCharCode;
```

Comment:
See how you like https://github.com/mishoo/UglifyJS2/pull/2236/commits/4e670b3d3dc1c8b7cd659ab05a66eeeb597e1ef1